src/google/cloud/ndb/query.py

```diff
@@ -1022,6 +1061,200 @@ def __init__(
+        # if post_filters is not None:
+        #     dsquery = datastore_query._AugmentedQuery(
+        #        dsquery, in_memory_filter=post_filters._to_filter(post=True)
+        #     )
+        return dsquery
+
+    @property
+    def is_distinct(self):
+        """True if results are guaranteed to contain a unique set of property
+      values.
+      This happens when every property in the group_by is also in the projection.
+      """
+        return bool(
+            self.group_by
+            and set(self._to_property_names(self.group_by))
+            <= set(self._to_property_names(self.projection))
+        )
+
+    def filter(self, *args):
+        """Return a new Query with additional filter(s) applied."""
+        if not args:
+            return self
+        preds = []
+        f = self.filters
+        if f:
+            preds.append(f)
+        for arg in args:
+            if not isinstance(arg, Node):
+                raise TypeError(
+                    "Cannot filter a non-Node argument; received %r" % arg
+                )
+            preds.append(arg)
+        if len(preds) == 1:
+            pred = preds[0]
+        else:
+            pred = ConjunctionNode(*preds)
+        return self.__class__(
+            kind=self.kind,
+            ancestor=self.ancestor,
+            filters=pred,
+            orders=self.orders,
+            app=self.app,
+            namespace=self.namespace,
+            default_options=self.default_options,
+            projection=self.projection,
+            group_by=self.group_by,
+        )
+
+    def order(self, *args):
+        """Return a new Query with additional sort order(s) applied."""
+        if not args:
+            return self
+        orders = self.orders
+        if orders is None:
+            orders = list(args)
+        else:
+            orders = list(orders)
+            orders.extend(args)
+        return self.__class__(
+            kind=self.kind,
+            ancestor=self.ancestor,
+            filters=self.filters,
+            orders=orders,
+            app=self.app,
+            namespace=self.namespace,
+            default_options=self.default_options,
+            projection=self.projection,
+            group_by=self.group_by,
+        )
+
+    def analyze(self):
+        """Return a list giving the parameters required by a query."""
+
+        class MockBindings(dict):
+            def __contains__(self, key):
+                self[key] = None
+                return True
+
+        bindings = MockBindings()
+        used = {}
+        ancestor = self.ancestor
+        if isinstance(ancestor, ParameterizedThing):
+            ancestor = ancestor.resolve(bindings, used)
+        filters = self.filters
+        if filters is not None:
+            filters = filters.resolve(bindings, used)
+        return sorted(used)  # Returns only the keys.
+
+    def bind(self, *args, **kwds):
+        """Bind parameter values.  Returns a new Query object."""
```

Comment:
Maybe something about what parameter binding is. Could go in class level docstring if you think that's appropriate.

Reply:
Also, what are the args?